servo/connectors/kubernetes.py

```diff
@@ -2698,7 +3290,7 @@ def container_name(self) -> str:
     # TODO: Factor into another class?
     async def _configure_tuning_pod_template_spec(self) -> None:
         # Configure a PodSpecTemplate for the tuning Pod state
-        pod_template_spec: kubernetes_asyncio.client.models.V1PodTemplateSpec = copy.deepcopy(self.deployment.pod_template_spec)
+        pod_template_spec: kubernetes_asyncio.client.models.V1PodTemplateSpec = await self.deployment.get_tuning_pod_template_spec()
```

Comment:
Tuning Pods are a concept that only exist in the canary mode. Pushing that concept up into the `Deployment` model is wrong because it isn't always a thing